crates/router/src/core/utils.rs

```diff
@@ -860,6 +870,156 @@ pub async fn construct_upload_file_router_data<'a>(
+        &mca.connector_name,
+        payment_attempt.merchant_connector_id.as_ref(),
+    )
+    .await?;
+
+    let test_mode = merchant_connector_account.is_test_mode_on();
+
+    let connector_auth_type: types::ConnectorAuthType = merchant_connector_account
+        .get_connector_account_details()
+        .parse_value("ConnectorAuthType")
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed while parsing value for ConnectorAuthType")?;
+
+    let add = payment_data.address.clone();
+
+    let shipping_address = payment_data
+        .tax_data
+        .clone()
+        .map(|tax_data| tax_data.shipping_details)
+        .clone()
+        .ok_or(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Missing shipping_details")?;
+
+    let order_details: Option<Vec<OrderDetailsWithAmount>> = payment_intent
+        .order_details
+        .clone()
+        .map(|order_details| {
+            order_details
+                .iter()
+                .map(|data| {
+                    data.to_owned()
+                        .parse_value("OrderDetailsWithAmount")
+                        .change_context(errors::ApiErrorResponse::InvalidDataValue {
+                            field_name: "OrderDetailsWithAmount",
+                        })
+                        .attach_printable("Unable to parse OrderDetailsWithAmount")
+                })
+                .collect::<Result<Vec<_>, _>>()
+        })
+        .transpose()?;
+
+    let router_data = types::RouterData {
+        flow: PhantomData,
+        merchant_id: merchant_account.get_id().to_owned(),
+        customer_id: payment_intent.customer_id.to_owned(),
+        connector_customer: None,
+        connector: mca.connector_name.clone(),
+        payment_id: payment_attempt.payment_id.clone(),
+        attempt_id: payment_attempt.attempt_id.clone(),
+        status: payment_attempt.status,
+        payment_method: diesel_models::enums::PaymentMethod::default(),
+        connector_auth_type,
+        description: payment_intent.description.clone(),
+        return_url: payment_intent.return_url.clone(),
+        address: add,
+        auth_type: payment_attempt.authentication_type.unwrap_or_default(),
+        connector_meta_data: None,
+        connector_wallets_details: None,
+        amount_captured: payment_intent
+            .amount_captured
+            .map(|amt| amt.get_amount_as_i64()),
+        access_token: None,
+        session_token: None,
+        reference_id: None,
+        payment_method_token: None,
+        recurring_mandate_payment_data: None,
+        preprocessing_id: None,
+        payment_method_balance: None,
+        connector_api_version: None,
+        request: types::PaymentsTaxCalculationData {
+            amount: payment_intent.amount,
+            shipping_cost: payment_intent
+                .shipping_cost
+                .ok_or(errors::ApiErrorResponse::InternalServerError)
```

Comment:
this should not be `InternalServerError`. Make it as `MissingRequiredField` or use `get_required_field` if this is mandatory.